src/category_theory/instances/Top/presheaf_of_functions.lean

```diff
@@ -0,0 +1,84 @@
+import category_theory.instances.Top.presheaf
+import category_theory.instances.TopCommRing.basic
+import category_theory.yoneda
+import group_theory.submonoid
+import ring_theory.subring
+import topology.algebra.continuous_functions
+
+universes v u
+
+open category_theory
+open category_theory.instances
+open topological_space
+
+open category_theory
+open category_theory.instances
+open topological_space
+
+namespace category_theory.instances.Top
+
+variables (X : Top.{v})
+
+def presheaf_to_Top (T : Top.{v}) : X.presheaf (Type v) :=
+(opens.to_Top X).op ⋙ (yoneda.obj T)
+
+-- TODO upgrade the result to TopCommRing?
+def continuous_functions (X : Top.{v}ᵒᵖ) (R : TopCommRing.{v}) : CommRing.{v} :=
+{ α := (unop X) ⟶ (TopCommRing.forget_to_Top.obj R),
+  str := _root_.continuous_comm_ring } -- but infer_instance doesn't work?
+
+namespace continuous_functions
+@[simp] def one (X : Top.{v}ᵒᵖ) (R : TopCommRing.{v}) (x) :
+  (monoid.one ↥(continuous_functions X R)).val x = 1 :=
+rfl
```

Comment:
use `lemma` and maybe put `rfl` on the same line. Also 2x below.

Reply:
Fixed.